tests/Menu/Integration/TabMenuTest.php

```diff
@@ -95,6 +95,18 @@ public function testLabelTranslationDomainOverride(): void
         $this->assertStringContainsString('my-other-translation', $html);
     }
 
+    public function testLabelAccessorErrorCase(): void
+    {
+        $translatorProphecy = $this->prophesize(TranslatorInterface::class);
+        $this->translator = $translatorProphecy->reveal();
+        $factory = new MenuFactory();
+        $menu = new MenuItem('test-menu', $factory);
+        $menu->addChild('some-label', ['uri' => '/whatever']);
+        $menu->getFirstChild()->addChild('label', ['uri' => '/whatever']);
+        $this->expectException(\TypeError::class);
```

Comment:
I didn't analyzed how to cover the fix and I don't know if is it feasible in an easy way; but what I've proposed is to ensure the issue is resolved with your changes. I guess the only way to do that is to render the template you've changed.